### PR TITLE
feat: polish select

### PR DESCRIPTION
**Pull request by zzmp**
- Replaces test JSON-RPC urls with our dev infura urls
- Filters selected token from the token select
- Loads native currency with the other chain tokens (smoother loading)
- Switches currencies when selecting the other currency

### CODE REVIEW

File: src/lib/components/TokenSelect/index.tsx [hunk 1]
```diff
@@ -17,14 +18,18 @@ const SearchInput = styled(StringInput)`
   ${inputCss}
 `
 
-export function TokenSelectDialog({ onSelect }: { onSelect: (token: Currency) => void }) {
+interface TokenSelectDialogProps {
+  value?: Currency
```
Comment (ianlapham) [inline src/lib/components/TokenSelect/index.tsx:22]: can this be called `currency`?

File: src/lib/components/TokenSelect/index.tsx [hunk 2]
```diff
@@ -81,7 +86,7 @@ export default function TokenSelect({ value, collapsed, disabled, onSelect }: To
       <TokenButton value={value} collapsed={collapsed} disabled={disabled} onClick={() => setOpen(true)} />
```
Comment (ianlapham) [inline src/lib/components/TokenSelect/index.tsx:86]: same as above - or `selectedCurrency` or something

File: src/lib/hooks/swap/index.ts [hunk 3]
```diff
@@ -45,3 +35,38 @@ export function useSwitchSwapCurrencies() {
...
+    },
+    [otherCurrency, setCurrency, switchCurrencies]
+  )
+  return [currency, setOrSwitchCurrency]
+}
+
+export function useSwapAmount(field: Field): [string | undefined, (amount: string) => void] {
+  const amount = useAtomValue(amountAtom)
+  const independentField = useAtomValue(independentFieldAtom)
+  const value = useMemo(() => (independentField === field ? amount : undefined), [amount, independentField, field])
```
Comment (ianlapham) [inline src/lib/hooks/swap/index.ts:61]: Ahh i like having this reset to undefined (not sure if it actually did that before)

**Review by ianlapham — APPROVED** [review]
Just small naming nits

File: src/lib/components/TokenSelect/index.tsx (hunk 1 above)
Comment (zzmp) [reply src/lib/components/TokenSelect/index.tsx:22]: I want to keep it as value to mirror native select elements, and make it obvious that this is the currently selected value.
I think all the tokens should be replaced with currency, but am waiting to do that until we've got some down time.

File: src/lib/components/TokenSelect/index.tsx (hunk 2 above)
Comment (zzmp) [reply src/lib/components/TokenSelect/index.tsx:86]: see above - `value` is intentional

**Pull request merged**